src/main/java/com/google/sps/utility/LocationsUtility.java

```diff
@@ -58,4 +59,27 @@ private static String getLocation(String prefix, String taskNotes) {
     }
     return "No " + prefix;
   }
+
+  /**
+   * Generates all permutations of the lists recursively. Scope of method is public for testing
+   * purposes.
+   *
+   * @param lists A list of lists to generate permutations for.
+   * @param result A pointer to the result since the method is recursive.
+   * @param depth An int which specifies the list in lists to traverse.
+   * @param current A list which represents the result for the current recursive stack.
+   */
+  public static void generateCombinations(
```

Comment:
If you must use a pointer for the result, I would suggest creating a wrapper class that calls this recursive function and then returns the result. 

i.e. Call `generateCombinations(lists)`, which then calls this recursive function (which should then be private), and then returns `result`

Reply:
Wrapper functions, good idea! Thanks!

Reply:
`RESOLVED`